src/handler/manager_line/setup.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  FollowEvent,
+  Message,
+  MessageEvent,
+  TemplateMessage,
+  TextEventMessage,
+  TextMessage,
+  WebhookEvent,
+} from '@line/bot-sdk'
+import { keyword } from '../../consts/keyword'
+import { ConfirmTemplate, TextTemplate } from '../../lib/line/template'
+import { Manager } from '../../types/managers'
+
+export const tellWelcome = (event: WebhookEvent) => {
+  return TextTemplate(
+    `友だち追加ありがとうございます。\nこのアカウントでは、文章や画像をチャット送っていただくだけで記事投稿が出来ます。`,
+  )
+}
+
+export const askName = () => {
+  const message: TextMessage = {
+    type: 'text',
+    text: '早速ですが、お名前を教えてください。※この情報はWebサイトには表示されません。',
+  }
+  return message
+}
+
+export const tellWelcomeBack = (manager: Manager) => {
+  const message: TextMessage = {
+    type: 'text',
+    text: `${manager.name}さん、お帰りなさい！`,
+  }
+  return message
+}
+
+export const confirmName = (event: MessageEvent) => {
+  if (event.message.type !== 'text') return
+  return ConfirmTemplate(`お名前は${event.message.text}でよろしいですか？`, '名前確認')
+}
```

Comment:
この層は、必要なパラメータを直接受け取りメッセージを整形するだけにした方がよくないですかね？
（前にも言いましたが、LINEのSDKの仕様`event.message.type`がここまで広がってしまうことになっています）
どんなケースで使われる/使われない、をここで管理すべきではないと考えています。
validationをやるなら、上位で済ませるべきかなと。
他の場所も同様です。

```suggestion
export const confirmName = (name: string) => {
  return ConfirmTemplate(`お名前は${name}でよろしいですか？`, '名前確認')
}
```

Reply:
分かりました。
validationが必要ない場合も、特定の情報しか必要としない場合はそれだけを渡す方が良いですか？

Reply:
はい。ここはプリミティブなメッセージ整形に止めて、名前をハメたい時は、stringを渡すようにしましょう。
修正コードもそうなっていますね。